src/middlewared/middlewared/plugins/docker/validation_utils.py

```diff
@@ -0,0 +1,45 @@
+import ipaddress
+
+from middlewared.schema import ValidationErrors
+
+
+def validate_address_pools(system_ips: list[dict], user_specified_networks: list[dict]):
+    verrors = ValidationErrors()
+    if not user_specified_networks:
+        verrors.add('docker_update.address_pools', 'At least one address pool must be specified')
+    verrors.check()
+
+    network_cidrs = set([
+        ipaddress.ip_network(f'{ip["address"]}/{ip["netmask"]}', False)
+        for ip in system_ips
+    ])
+    seen_networks = set()
+    for index, user_network in enumerate(user_specified_networks):
+        base_network = ipaddress.ip_network(user_network['base'], False)
+        subnet_prefix = int(user_network['base'].split('/')[-1])
```

Comment:
This is the subnet suffix based on taking last entry after `split()`. Furthermore, you shouldn't need to do this since `base_network` should have the information?